src/input/type_record.cc

```diff
@@ -634,6 +637,44 @@ unsigned int AbstractRecord::child_size() const {
 }
 
 
+int AbstractRecord::add_child(const Record &subrec)
+{
+	if ( !TypeBase::was_constructed(&subrec) ) {
+		xprintf(Warn, "Add non-constructed record '%s' to abstract record '%s'!\n", subrec.type_name().c_str(), this->type_name().c_str());
+	}
+	add_descendant(subrec);
+	//subrec.make_copy_keys(*this);
+
+	return 1;
+}
+
+
+bool AbstractRecord::finish() {
+	if (data_->finished) return true;
+
+	ASSERT(data_->closed_, "Finished AbstractRecord '%s' must be closed!", this->type_name().c_str());
+
+	data_->finished = true;
+	no_more_descendants();
+
+	return (data_->finished);
+}
+
+
+AbstractRecord &AbstractRecord::close() {
+	data_->closed_=true;
+    return *( Input::TypeRepository<AbstractRecord>::getInstance().add_type( *this ).get() );
```

Comment:
Why you are still using explicit call of get()?

return *( Input::TypeRepository<AbstractRecord>::getInstance().add_type( *this ));
.. should work as well.